src/main/java/ArrayIterator.java

```diff
@@ -0,0 +1,30 @@
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class ArrayIterator<T> implements Iterator<T> {
+
+    private int size;
+    private int currentIndex;
+    private T[] items;
+
+    public ArrayIterator(T[] items, int size) {
+        this.items = items;
+        this.size = size;
+        this.currentIndex = 0;
+    }
+
+    @Override
+    public boolean hasNext() {
+        return currentIndex < size;
+    }
+
+    @Override
+    public T next() {
+        if (!hasNext()) {
+            throw new NoSuchElementException();
+        }
+        T returnElement = items[currentIndex];
+        currentIndex--;
+        return returnElement;
+    }
+}
```

Comment:
Run the formatter? Missing blank line at the end.

If you're using intellij, download the [save actions](https://plugins.jetbrains.com/plugin/7642-save-actions) plugin, then use that to format whenever the file saves.